core/src/main/java/tech/tablesaw/io/ReadOptions.java

```diff
@@ -294,6 +301,15 @@ public Builder minimizeColumnSizes() {
       return this;
     }
 
+    /**
+     * If the files contains more than numSamples, performs sampling using the single-pass reservoir
+     * sampling algorithm (https://en.wikipedia.org/wiki/Reservoir_sampling).
+     */
+    public Builder sampleRowsIfGreaterThan(int numSamples) {
```

Comment:
This name's a bit long. Can we come up with something shorter? How about `sampleSize`?

Reply:
Sound good, but if the file has fewer rows than `numSamples`, the actual sample size will be smaller than `numSamples`. Does `maxSampleSize` sounds more descriptive for you?

Reply:
Really? Can you explain more? I thought that if we passed in `numSamples` here that was greater than the number of rows in the file then the entire file would be read into the table

Reply:
Yes. If `numSamples` is smaller than the number of rows in the file than exactly `numSamples` are returned. If the file has `n` rows, and and `numSamples` > `n`, then `n` random samples are returned. Therefore, `numSamples` is the maximum number of samples returned since it is possible the number of rows in the file is less than `numSamples`.

Reply:
Ok, I think we had different connotations in mind for "samples". In my mind, there's no sampling if all records are returned, which is why `sampleSize` makes sense to me. `maxSampleSize` to me sounds confusing because it makes it sound like there might be sampling, but it's not clear how much.

Reply:
You're right, effectively there is no sampling, but there could be oversampling to increase the samples to match the selected sample size. Nevertheless, I'm fine with `sampleSize`. I changed the code accordingly.